src/runtime-rs/crates/hypervisor/src/device/topology.rs

```diff
@@ -354,13 +356,32 @@ impl PCIeTopology {
         let total_rp = topo_config.device_info.pcie_root_port;
         let total_swp = topo_config.device_info.pcie_switch_port;
 
+        let hot_plug = match topo_config.device_info.hot_plug_vfio.as_deref() {
+            Some("root-port") => PCIePort::RootPort,
+            Some("switch-port") => PCIePort::SwitchPort,
+            _ => PCIePort::NoPort,
+        };
+        
+        let cold_plug = match topo_config.device_info.cold_plug_vfio.as_deref() {
+            Some("root-port") => PCIePort::RootPort,
+            Some("switch-port") => PCIePort::SwitchPort,
+            _ => PCIePort::NoPort,
+        };
+
+        let (hot_plug, cold_plug) = if hot_plug != PCIePort::NoPort && cold_plug != PCIePort::NoPort {
+            (hot_plug, PCIePort::NoPort)
+        } else {
+            (hot_plug, cold_plug)
+        };
```

Comment:
Error out instead, as that makes it explicit for the user that it's a config error.